sycl/test-e2e/Basic/built-ins/marray_relational.cpp

```diff
@@ -25,6 +25,25 @@
     }                                                                          \
   }
 
+#define TEST2(FUNC, TYPE, EXPECTED, N, ...)                                    \
+  {                                                                            \
+    {                                                                          \
+      TYPE result[1];                                                          \
+      {                                                                        \
+        sycl::buffer<TYPE> b(result, sycl::range{1});                          \
+        deviceQueue.submit([&](sycl::handler &cgh) {                           \
+          sycl::accessor res_access{b, cgh};                                   \
+          cgh.single_task([=]() {                                              \
+            int res = FUNC(__VA_ARGS__);                                       \
+            for (int i = 0; i < N; i++)                                        \
+              res_access[0] = res;                                             \
+          });                                                                  \
+        });                                                                    \
+      }                                                                        \
+      assert(result[0] == EXPECTED[0]);                                        \
```

Comment:
I'd use

```
buffer<TYPE> b{1};
q.submit();
assert(host_accessor{b}[0] == EXPECTED[0]);
```

to reduce indentation.

Reply:
All these marray***.cpp tests will be refactored to get rid of macros in the separate patch, this comment will be applied.